core/src/main/scala/org/apache/spark/util/SignalUtils.scala

```diff
@@ -51,11 +88,8 @@ private[spark] object Signaling extends Logging {
       // register old handler, will receive incoming signals while this handler is running
       Signal.handle(signal, prevHandler)
 
-      val escalate = actions.asScala forall { action =>
-        !action()
-      }
-
-      if(escalate) {
+      val escalate = actions.asScala.forall { action => !action() }
```

Comment:
sure, I initially had comments but thought they were overkill since the behaviour can be deducted from the comments in the register() method. I'll add some more info